pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -209,6 +217,71 @@ func (n *clusterManagerController) sync(ctx context.Context, controllerContext f
 	}
 	managementClient := n.operatorKubeClient // We assume that operator is always running on the management cluster.
 
+	if config.MWReplicaSetEnabled {
+		// If the work driver is not kube, we need to sync the work driver config secret
+		if workDriver != operatorapiv1.WorkDriverTypeKube {
+			operatorNamespace, err := helpers.GetOperatorNamespace()
+			if err != nil {
+				return err
+			}
+			// check the secret containing work driver config explicitly because
+			// resourceapply.SyncSecret below won't return err if the source secret is not found
+			_, err = n.operatorKubeClient.CoreV1().Secrets(operatorNamespace).Get(ctx, helpers.WorkDriverConfig, metav1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					klog.Errorf("Secret %s not found in namespace %s", helpers.WorkDriverConfig, operatorNamespace)
+					conditionChanged := meta.SetStatusCondition(&clusterManager.Status.Conditions, metav1.Condition{
+						Type:    workDriverConfigSecretSynced,
+						Status:  metav1.ConditionFalse,
+						Reason:  "WorkDriverConfigSecretNotFound",
+						Message: "Work driver config secret is not found",
+					})
+					if conditionChanged {
+						clusterManager.Status.ObservedGeneration = clusterManager.Generation
+						_, err := n.patcher.PatchStatus(ctx, clusterManager, clusterManager.Status, originalClusterManager.Status)
+						if err != nil {
+							return err
+						}
+					}
+				}
+				return err
+			} else {
```

Comment:
to many nested if. should refactor the code here. 
Wouldn't SyncSecret return a notFound error? I think we do not need to get then sync.

directly sync and return error should be ok.

Reply:
SyncSecret doesn't return an error when the source secret doesn't exist, that's why I added the check before SyncSecret. Also added comments at https://github.com/open-cluster-management-io/ocm/blob/a559f7eeb1ec7bbf4041ceed029cafbddd20d796/pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go#L227-L229